File: C#Advanced/Exams/StringMatrixRotation/StringMatrixRotation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace StringMatrixRotation
{
    class StringMatrixRotation
    {
        static void Main(string[] args)
        {
            string input;
            var maxLength = 0;
            var degreesString = Console.ReadLine();
            degreesString = Regex.Replace(degreesString, @"[^0-9]", "");
            int degrees = int.Parse(degreesString) % 360;
            List<string> lines = new List<string>();

            while ((input = Console.ReadLine()) != "END")
            {
                lines.Add(input);
            }
            for (var i = 0; i < lines.Count; i++)
            {
                while (lines[i].Length > maxLength)
                {
                    maxLength = lines[i].Length;
                }
            }
            char[,] matrix = buildMatrix(lines, lines.Count, maxLength);
            switch (degrees % 360)
            {
                case 0: PrintMatrix(matrix); break;
                case 90: Rotate90(matrix); break;
                case 180: Rotate180(matrix); break;
                case 270: Rotate270(matrix); break;
            }
        }

        static char[,] buildMatrix(List<string> source, int N, int M)
        {
            var result = new char[N, M];
            for (int i = 0; i < N; ++i)
            {
                for (int j = 0; j < M; ++j)
                {
                    result[i, j] = source[i].PadRight(M, ' ')[j];
                }
            }
            return result;
        }

        private static void PrintMatrix(char[,] matrix)
        {
            for (int row = 0; row < matrix.GetLength(0); row++)
            {
                for (int col = 0; col < matrix.GetLength(1); col++)
                {
                    Console.Write(matrix[row, col]);
                }
                Console.WriteLine();
            }
        }

        private static void Rotate90(char[,] matrix)
        {
            for (int rows = 0; rows < matrix.GetLength(1); rows++)
            {
                for (int cols = matrix.GetLength(0) - 1; cols >= 0; cols--)
                {
                    Console.Write(matrix[cols, rows]);
                }
                Console.WriteLine();
            }
        }

        private static void Rotate180(char[,] matrix)
        {
            for (int rows = matrix.GetLength(0) - 1; rows >= 0; rows--)
            {
                for (int cols = matrix.GetLength(1) - 1; cols >= 0; cols--)
                {
                    Console.Write(matrix[rows, cols]);
                }
                Console.WriteLine();
            }
        }

        private static void Rotate270(char[,] matrix)
        {
            for (int rows = matrix.GetLength(1) - 1; rows >= 0; rows--)
            {
                for (int cols = 0; cols < matrix.GetLength(0); cols++)
                {
                    Console.Write(matrix[cols, rows]);
                }
                Console.WriteLine();
            }
        }
    }
}

File: JavaScriptBasics/Exams/Exam-24-November-2014/Problem-4-User-Module-Vladislav-Hadjiiski/Problem-4_Dancho/ExamProblems/ExamProblems/StringMatrixRotation.js
function solve(args) {
    var degrees = args[0].split(/\D+/)[1] % 360;
    var matrix = [];
    var len = args.length;
    var maxLength = 0;
    for (var i = 1; i < len; i += 1) {
        matrix[i - 1] = args[i].split("");
        if (args[i].length > maxLength) {
            maxLength = args[i].length;
        }
    }

    len = matrix.length;
    for (var row = 0; row < len; row += 1) {
        for (var col = matrix[row].length; col < maxLength; col++) {
            matrix[row].push(" ");
        }
    }

    rotateMatrix(matrix, degrees);

    function rotateMatrix(matrix, degrees) {
        var row, col;
        switch (degrees) {
            case 0:
                for (row = 0; row < matrix.length; row++) {
                    var rowAsString = matrix[row].join("");
                    console.log(rowAsString);
                }
                break;
            case 90:
                for (col = 0; col < matrix[0].length; col++) {
                    var rowAsString = "";
                    for (row = 0; row < matrix.length; row++) {
                        rowAsString += matrix[row][col];
                    }
                    console.log(rowAsString.split("").reverse().join(""));
                }
                break;
            case 180:
                for (row = matrix.length - 1; row >= 0; row--) {
                    var rowAsString = matrix[row].reverse().join("");
                    console.log(rowAsString);
                }
                break;
            case 270:
                for (col = matrix[0].length - 1; col >= 0; col--) {
                    var rowAsString = "";
                    for (row = 0; row < matrix.length; row++) {
                        rowAsString += matrix[row][col];
                    }
                    console.log(rowAsString);
                }
                break;
        }
    }
}
